draft-ietf-curdle-gss-keyex-sha2.xml

```diff
@@ -176,201 +176,70 @@
-            the same way it is done in step 1 and peforms
-            the following computations:</t>
-        <t><list>
-          <t>K a shared secret obtained using ECDH key exchange:</t>
-          <t>
-          <list><t>Both client and server perform the same calculation where
-                  d_U is the secret value, d_C for client and d_S for server
-                  and q_V is the received public value, q_S for client and q_C
-                  for server.</t>
-              <t>For NIST curves, the peers calculate the shared secret field
-                  element using d_U and q_V using the algorithm in Section
-                  3.3.1 of <xref target="SEC1v2"/>. If the result is
-                  "invalid" (the shared secret point is a point
-                  at infinity), the key exchange MUST fail.</t>
-              <t>For NIST curves, the shared secret is the octet string
-                  representation of the field element using the algorithm
-                  described in Section 2.3.5 of <xref target="SEC1v2"/>.</t>
-              <t>For curve25519 and curve448, the peers apply the X25519 or
-                  X448 function, respectively for curve25519 and curve448,
-                  on the d_U and q_V. The result of the function is
-                  the shared secret.</t>
-              <t>For curve25519 and curve448, if all the octets of the shared
-                  secret are zero octets, the key exchange MUST fail.</t>
-          </list>
-         </t>
-
-          <t>To verify integrity of the hanshake, peers use the HASH to
-             calculate H:</t>
-
-          <t>H = hash(V_C || V_S || I_C || I_S || K_S || Q_C || Q_S ||
-          K).</t>
-
-          <t>MIC is the GSS-API message integrity code for H computed by
-          calling GSS_GetMIC().</t>
-
-          <t>S then sends Q_S and the message integrity code (MIC) to C.</t>
-        </list>
+        be set to true if the client wishes to hide its identity.
+        This key exchange process will exchange only a single token once the
+        context has been established, therefore the replay_det_req_flag and
+        sequence_req_flag SHOULD be set to "false".</t>
+
+        <t>The client MUST include its public key with the first message it
+        sends to the server during this process; if the server receives
+        more than one key or none at all, the key exchange MUST fail.</t>
+
+        <t>During GSS Context estalishment multiple tokens may be exchanged by
+        the client and the server. When the GSS Context is established
+        (major_status is GSS_S_COMPLETE) the parties check that
+        mutual_state and integ_avail are both "true". If not the key
+        exchange MUST fail.</t>
+
+        <t>Once a party receives the peer's public key it proceeds to compute
+        a shared secret K according to Section 3.3.1 of <xref target="SEC1v2"/>
+        (where it is called 'z') for NIST Curves or Section 6 of
```

Comment:
it is important to say _how_ the conversion happens, as the zeros at the beginning must be retained, and the number must be encoded as a big-endian integer

for x25519 and x448 it's easy because they output a byte string